### PR TITLE
Config object accessible by simulation class

### PR DESCRIPTION
**Pull request by andrewfullard**
## Description

Adds config object to simulation class

## Motivation and Context

Needed to output to HDF

## How Has This Been Tested?

Standard unit tests pass as expected

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

File: tardis/simulation/base.py [hunk 1]
```diff
@@ -109,6 +109,7 @@ class Simulation(PlasmaStateStorerMixin, HDFWriterMixin):
         "iterations_t_rad",
         "iterations_electron_densities",
         "iterations_t_inner",
+        "config"
```
Comment (marxwillia) [inline tardis/simulation/base.py:112]: The config is not yet a supported hdf property.

Comment (wkerzendorf) [reply tardis/simulation/base.py:112]: yep - need to rebase

**Comment by andrewfullard** [comment]
Note- dependent in #1299

**Comment by andrewfullard** [comment]
Rebased and the simulation HDF contains the configuration in the `simulation/config/config` group. May need to be adjusted to just be under `simulation/config`.

**Comment by andrewfullard** [comment]
After testing I'm not sure this PR is needed. The configuration gets output in the main HDF regardless.

**Pull request closed without merging**